File: questao_03/questao_03.java
class ContaBancaria {
    private double saldo;
    @SuppressWarnings("unused")
    private String titular;

    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.printf("Deposito de R$ %.2f realizado. Saldo atual: R$ %.2f%n", valor, saldo);
        } else {
            System.out.println("Valor de depósito deve ser positivo.");
        }
    }

    public void sacar(double valor) {
        if (valor > saldo) {
            System.out.println("Saldo insuficiente para realizar o saque.");
        } else {
            saldo -= valor;
            System.out.printf("Saque de R$ %.2f realizado. Saldo atual: R$ %.2f%n", valor, saldo);
        }
    }

    public double getSaldo() {
        return saldo;
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("Arthur", 1000);

        conta.depositar(200);
        conta.sacar(150);
        conta.sacar(1200);
        System.out.printf("Saldo final: R$ %.2f%n", conta.getSaldo());
    }
}

File: questao_03/questao_03.go
package main

import (
	"fmt"
)

type ContaBancaria struct {
	titular string
	saldo   float64
}


func (c *ContaBancaria) Depositar(valor float64) {
	if valor > 0 {
		c.saldo += valor
		fmt.Printf("Deposito de R$ %.2f realizado. Saldo atual: R$ %.2f\n", valor, c.saldo)
	} else {
		fmt.Println("Valor de depósito deve ser positivo.")
	}
}


func (c *ContaBancaria) Sacar(valor float64) {
	if valor > c.saldo {
		fmt.Println("Saldo insuficiente para realizar o saque.")
	} else {
		c.saldo -= valor
		fmt.Printf("Saque de R$ %.2f realizado. Saldo atual: R$ %.2f\n", valor, c.saldo)
	}
}


func (c ContaBancaria) GetSaldo() float64 {
	return c.saldo
}

func main() {
	conta := ContaBancaria{titular: "Arthur", saldo: 1000}

	conta.Depositar(200)
	conta.Sacar(150)
	conta.Sacar(1200)
	fmt.Printf("Saldo final: R$ %.2f\n", conta.GetSaldo())
}
